lib/constructor.js

```diff
@@ -253,6 +253,56 @@ util.inherits(Sharp, stream.Duplex);
  * // firstWritableStream receives auto-rotated, resized readableStream
  * // secondWritableStream receives auto-rotated, extracted region of readableStream
  *
+ * @example
+ * // Create a pipeline that will download an image, resize it and format it to different files
+ * // Using Promises to know when the pipeline is complete
+ * const fs = require("fs");
+ * const got = require("got");
+ * const sharpStream = sharp({
+ *   failOnError: false
+ * });
+ *
+ * const promises = [];
+ *
+ * promises.push(
+ *   sharpStream
+ *     .clone()
+ *     .jpeg({ quality: 100 })
+ *     .toFile("originalFile.jpg")
+ * );
+ *
+ * promises.push(
+ *   sharpStream
+ *     .clone()
+ *     .resize({ width: 500 })
+ *     .jpeg({ quality: 80 })
+ *     .toFile("optimized-500.jpg")
+ * );
+ *
+ * promises.push(
+ *   sharpStream
+ *     .clone()
+ *     .resize({ width: 500 })
+ *     .webp({ quality: 80 })
+ *     .toFile("optimized-500.webp")
+ * );
+ *
+ * // https://github.com/sindresorhus/got#gotstreamurl-options
+ * got.stream("https://www.example.com/some-file.jpg").pipe(sharpStream);
+ *
+ * fs.createReadStream(jpgOriginalImagePath).pipe(sharpStream);
```

Comment:
Might the piping of two streams into the same sharp instance confuse people? I'd prefer to stick to one example, perhaps focus on the use of `got` as the use of HTTP Stream-based input is more common (for filesystem-based input, passing the path to the constructor is preferable).

Reply:
ah! that's a mistake of me, in my code those two (got/fs.createReadStream) are in different code paths (if ...), Ill update that! (definitely confusing lol)

Reply:
ok just updated the pr now, thx!